app/scripts/components/analysis/define/use-stac-collection-search.ts

```diff
@@ -37,10 +33,31 @@ export function useStacCollectionSearch({
   const result = useQuery({
     queryKey: ['stacCollection'],
     queryFn: async ({ signal }) => {
-      const collectionResponse = await axios.get(collectionUrl, {
-        signal
-      });
-      return collectionResponse.data.collections;
+      const collectionUrlsFromDataSets = allAvailableDatasetsLayers
+        .filter((dataset) => dataset.stacApiEndpoint)
+        .map((dataset) => `${dataset.stacApiEndpoint}/collections`)
+        .filter((value, index, array) => array.indexOf(value) === index);
+
+      const collectionUrls = [
+        ...collectionUrlsFromDataSets,
+        `${process.env.API_STAC_ENDPOINT}/collections`
+      ];
+
+      const colloectionRequests = collectionUrls.map((url: string) =>
+        axios.get(url, { signal }).then((response) => {
+          return response.data.collections.map((col) => ({
+            ...col,
+            stacApiEndpoint: url.replace(/\/collections$/, '')
+          }));
+        })
+      );
+      return axios.all(colloectionRequests).then(
+        axios.spread((...responses) => {
+          // Merge all responses into one array
+          const mergedData = [].concat(...responses);
+          return mergedData;
+        })
+      );
```

Comment:
⛏️  : can we possibly simplify here with 
`axios.spread((...responses) => [].concat(...responses))`
?